velox/type/TimestampConversion.cpp

```diff
@@ -268,8 +268,16 @@ bool tryParseDateString(
     return false;
   }
 
-  // In standard-cast mode, no more trailing characters.
   if (mode == ParseMode::kStandardCast) {
+    // Skip trailing spaces.
+    while (pos < len && characterIsSpace(buf[pos])) {
+      pos++;
+    }
+    // Check position. if end was not reached, non-space chars remaining.
+    if (pos < len) {
+      return false;
+    }
+
```

Comment:
@marin-ma Thanks for making all the changes till now. I have a last request, can you please move this part of the change out in a separate PR? This unfortunately, is a change in behavior of existing `cast` and can result in issues for existing workloads that dont expect this behavior. It would be good to separate it in case we need to roll back this behavior change while still keeping all the spark specific functionality you added here.